mlrun/projects/project.py

```diff
@@ -35,50 +32,117 @@
+        project.set_function('prep_data.py', 'prep-data', image='mlrun/mlrun', handler='prep_data')
+        project.set_function('hub://sklearn_classifier', 'train')
+        project.set_artifact('data', Artifact(target_path=data_url))
+        project.set_workflow('main', "./myflow.py")
+        project.save()
+
+        # run the "main" workflow (watch=True to wait for run completion)
+        project.run("main", watch=True)
+
+    example (load from template)::
+
+        # create a new project from a zip template (can also use yaml/git templates)
+        # initialize a local git, and register the git remote path
+        project = mlrun.new_project("myproj", "./", init_git=True,
+                                    remote="git://github.com/mlrun/project-demo.git",
+                                    from_template="http://mysite/proj.zip")
+        project.run("main", watch=True)
+
 
     :param name:         project name
     :param context:      project local directory path
     :param init_git:     if True, will git init the context dir
     :param user_project: add the current user name to the provided project name (making it unique per user)
+    :param remote:       remote Git url
+    :param from_template:     path to project YAML/zip file that will be used as from_template
+    :param secrets:      key:secret dict or SecretsStore used to download sources
+    :param description:  text describing the project
 
     :returns: project object
     """
-    if user_project:
-        user = environ.get("V3IO_USERNAME") or getpass.getuser()
-        name = f"{name}-{user}"
-
-    project = MlrunProject(name=name)
+    context = context or "./"
+    name = _add_username_to_project_name_if_needed(name, user_project)
+
+    if from_template:
+        if from_template.endswith(".yaml"):
+            project = _load_project_file(from_template, name, secrets)
+        elif from_template.startswith("git://"):
+            clone_git(from_template, context, secrets, clone=True)
+            shutil.rmtree(path.join(context, ".git"))
+            project = _load_project_dir(context, name)
+        elif from_template.endswith(".zip"):
+            clone_zip(from_template, context, secrets)
+            project = _load_project_dir(context, name)
+        else:
+            raise ValueError("template must be a path to .yaml or .zip file")
+        project.metadata.name = name
+    else:
+        project = MlrunProject(name=name)
     project.spec.context = context
 
-    if init_git:
-        repo = Repo.init(context)
-        project.spec.repo = repo
-
+    repo, url = init_repo(context, remote, init_git or remote)
+    project.spec.repo = repo
+    if remote and url != remote:
+        project.create_remote(remote)
+    elif url:
+        project.spec._source = url
+        project.spec.origin_url = url
+    if description:
+        project.spec.description = description
+    mlrun.mlconf.default_project = project.metadata.name
```

Comment:
The interaction between this and `set_environment` becomes a bit complex here - I think the idea was that if I did `set_environment` and set a project name (with or without username), then `new_project` can skip getting a name and in this case will use the `default_project` as project name.
This code actually works the opposite way - if you call `set_environment` then this overrides what you did. While it does make sense and these are different flows (use this for creating project, use `set_environment` for working with existing project) we should at least make sure we properly document these differences.

Reply:
we want to avoid the set_environment in future, always work in a project context as a practice, user start the NB w new/get/load project ..  there is no sense in creating a project and having the default point to another one, also we want the name to be explicit in new_project